File: src/main/java/gargoyle/ct/convert/impl/DoubleConverter.java
package gargoyle.ct.convert.impl;

import gargoyle.ct.convert.Converter;
import org.jetbrains.annotations.NotNull;

public class DoubleConverter implements Converter<Double> {
    @Override
    public String format(Double data) {
        return String.valueOf(data);
    }

    @NotNull
    @Override
    public Double parse(@NotNull String data) {
        return Double.valueOf(data);
    }
}

File: src/main/kotlin/gargoyle/ct/convert/impl/DoubleConverter.kt
package gargoyle.ct.convert.impl

import gargoyle.ct.convert.Converter

class DoubleConverter : Converter<Double> {
    override fun format(data: Double): String {
        return data.toString()
    }

    override fun parse(data: String): Double {
        return data.toDouble()
    }
}
